mongo/integration/client_test.go

```diff
@@ -768,6 +768,88 @@ func TestClient(t *testing.T) {
 				"expected 'OP_MSG' OpCode in wire message, got %q", pair.Sent.OpCode.String())
 		}
 	})
+
+	opts := mtest.NewOptions().
+		// Blocking failpoints don't work on pre-4.2 and sharded clusters.
+		Topologies(mtest.Single, mtest.ReplicaSet).
+		MinServerVersion("4.2").
+		// Expliticly enable retryable reads and retryable writes.
+		ClientOptions(options.Client().SetRetryReads(true).SetRetryWrites(true))
+	mt.RunOpts("operations don't retry after a context timeout", opts, func(mt *mtest.T) {
+		testCases := []struct {
+			desc      string
+			failPoint mtest.FailPoint
+			operation func(context.Context, *mongo.Collection) error
+		}{
+			{
+				desc: "read op",
+				failPoint: mtest.FailPoint{
+					ConfigureFailPoint: "failCommand",
+					Mode:               "alwaysOn",
+					Data: mtest.FailPointData{
+						FailCommands:    []string{"find"},
+						BlockConnection: true,
+						BlockTimeMS:     500,
+					},
+				},
+				operation: func(ctx context.Context, coll *mongo.Collection) error {
+					return coll.FindOne(ctx, bson.D{}).Err()
+				},
+			},
+			{
+				desc: "write op",
+				failPoint: mtest.FailPoint{
+					ConfigureFailPoint: "failCommand",
+					Mode:               "alwaysOn",
+					Data: mtest.FailPointData{
+						FailCommands:    []string{"insert"},
+						BlockConnection: true,
+						BlockTimeMS:     500,
+					},
+				},
+				operation: func(ctx context.Context, coll *mongo.Collection) error {
+					_, err := coll.InsertOne(ctx, bson.D{})
+					return err
+				},
+			},
+		}
+
+		for _, tc := range testCases {
+			mt.Run(tc.desc, func(t *mtest.T) {
```

Comment:
```suggestion
			mt.Run(tc.desc, func(mt *mtest.T) {
```

Reply:
Good catch! I actually changed the default param name to `mt` in https://github.com/mongodb/mongo-go-driver/pull/1589, but auto-completed that function signature before rebasing it on v1, so it got the old name. Will fix.